### PR TITLE
Health regeneration

### PR DESCRIPTION
**Pull request by fegbert**
Implemented health regeneration

### CODE REVIEW

File: src/main/kotlin/com/msd/robot/domain/Robot.kt [hunk 1]
```diff
@@ -191,4 +191,10 @@ class Robot(
             energyRegen
         }
     }
+
+    fun regenerateHealth(amount: Int) {
```
Comment (DennisWae) [inline src/main/kotlin/com/msd/robot/domain/Robot.kt:195]: According to the [wiki](https://the-microservice-dungeon.github.io/docs/rules/trading), robots can only be repaired to max health. Also please document your methods.

**Comment by DennisWae** [comment]
You should also write tests for the RobotApplicationService and RobotDomainService, but not implement them yet as someone else shoud do that.

**Review by DennisWae — CHANGES_REQUESTED** [review]

File: src/main/kotlin/com/msd/robot/domain/Robot.kt (hunk 1 above)
Comment (DennisWae) [reply src/main/kotlin/com/msd/robot/domain/Robot.kt:195]: This method should also probably be renamed to `repairRobot`.

File: src/main/kotlin/com/msd/robot/domain/Robot.kt [hunk 2]
```diff
@@ -191,4 +191,10 @@ class Robot(
...
+        if (amount + this.health > this.maxHealth) throw HealthFullException("Tried to regenerate $amount health but is already at full health.");
```
Comment (baldhat) [inline src/main/kotlin/com/msd/robot/domain/Robot.kt:196]: The robot should still heal up to its max health. This makes it impossible to heal up to 20, if the robot has 17 Health and tries to heal by 4

**Review by baldhat — CHANGES_REQUESTED** [review]

**Review by DennisWae — APPROVED** [review]

**Review by baldhat — APPROVED** [review]

**Pull request merged**